File: 2093-check-if-string-is-a-prefix-of-array/2093-check-if-string-is-a-prefix-of-array.java
class Solution {
    public boolean isPrefixString(String s, String[] words) {
        String prifix="";
        for(int i=0;i<words.length;i++){
            prifix+=words[i];
           if (prifix.equals(s)) {
                return true;
            }
            if(prifix.length()>s.length()){
                return false;
            }
        }
        return false;
    }
}

File: 2093-check-if-string-is-a-prefix-of-array/2093-check-if-string-is-a-prefix-of-array.py
class Solution:
    def isPrefixString(self, s: str, words: List[str]) -> bool:
        prifix=""
        for w in words:
            prifix+=w
            if prifix==s:
                return True 
            elif len(prifix)>len(s):
                return False
        return False
